gateway/dataimpl/server_v1/helpers.go

```diff
@@ -134,7 +134,7 @@ func bucketTypeFromCbmgmtx(t cbmgmtx.BucketType) (admin_bucket_v1.BucketType, *s
 		return admin_bucket_v1.BucketType_BUCKET_TYPE_EPHEMERAL, nil
 	}
 
-	return admin_bucket_v1.BucketType(0), status.New(codes.InvalidArgument, "invalid bucket type received")
+	return admin_bucket_v1.BucketType(0), status.New(codes.Internal, "invalid bucket type received")
```

Comment:
Is an invalid bucket type not actually an invalid argument? Same goes for several of these.

Reply:
Am I understanding correctly that here, there's a failure to understand something from the server?  In that case, I think Internal is the more reasonable code.  I feel InvalidArgument implies the user got something wrong.